backend/src/routes/spotify.ts

```diff
@@ -0,0 +1,180 @@
+import express from "express"
+import querystring from "querystring"
+import cors from "cors"
+
+import { generateRandomString } from "../utils";
+
+const clientSecret = process.env.SPOTIFY_CLIENT_SECRET || "";
+const clientId = process.env.SPOTIFY_CLIENT_ID || "";
+const redirectURI = "http://localhost:8000/api/spotify/callback";
+const frontendURI = "http://localhost:3000/"
+
+const router = express.Router();
+
+// Creates or regenerate session. This is so we can start fresh new when
+// going through the authentication process
+router.get("/session", (req, res) => {
+  req.session.regenerate((err) => {
+    if(err) {
+      console.log(`Cannot regenerate session`)
+      console.log(err)
+  
+      res.status(500);
+      res.send(`Cannot regenerate session ${err}`)
+    } else {
+      res.status(200);
+      res.send(req.session.id)
+    }
+
+  });
+
+})
+
+// Deletes session
+router.delete("/session", (req, res) => {
+  const sessionId = req.session.id;
+
+  req.session.destroy((err) => {
+    if(err) {
+      console.log(`Cannot regenerate session`)
+      console.log(err)
+
+      res.status(500);
+      res.send(`Cannot delete session ${err}`)
+    } else {
+      res.status(200);
+      res.send(`Deleted Session ${sessionId}`);
+    }
+  })
+
+})
+
+router.get("/login", (req, res) => {
+  const state = generateRandomString(16);
+  const scope = "user-read-private user-read-email streaming user-read-playback-state user-modify-playback-state";
```

Comment:
Could you please add a comment that these are the params for Spotify to specify the authorization scope? For my (and other people's) reference later.

I think it'd be good to also link the actual [API site](https://developer.spotify.com/documentation/general/guides/authorization/scopes/) for the scopes.